crates/evm/coverage/src/analysis.rs

```diff
@@ -454,30 +454,34 @@ impl<'a> ContractVisitor<'a> {
     /// collection (plus additional coverage line if item is a statement).
     fn push_item_kind(&mut self, kind: CoverageItemKind, src: &ast::LowFidelitySourceLocation) {
         let item = CoverageItem { kind, loc: self.source_location_for(src), hits: 0 };
-        // Push a line item if we haven't already
-        if matches!(item.kind, CoverageItemKind::Statement | CoverageItemKind::Branch { .. }) &&
-            self.last_line < item.loc.line
-        {
+
+        // Push a line item if we haven't already.
+        debug_assert!(!matches!(item.kind, CoverageItemKind::Line));
+        if self.last_line < item.loc.lines.start {
             self.items.push(CoverageItem {
                 kind: CoverageItemKind::Line,
                 loc: item.loc.clone(),
                 hits: 0,
             });
-            self.last_line = item.loc.line;
+            self.last_line = item.loc.lines.start;
         }
 
         self.items.push(item);
     }
 
     fn source_location_for(&self, loc: &ast::LowFidelitySourceLocation) -> SourceLocation {
-        let loc_start =
-            self.source.char_indices().map(|(i, _)| i).nth(loc.start).unwrap_or_default();
+        let bytes_start = loc.start as u32;
+        let bytes_end = (loc.start + loc.length.unwrap_or(0)) as u32;
+        let bytes = bytes_start..bytes_end;
+
+        let start_line = self.source[..bytes.start as usize].lines().count() as u32;
```

Comment:
there was a problem for running coverage on  windows https://github.com/foundry-rs/foundry/issues/8957 which was solved by  https://github.com/foundry-rs/foundry/pull/8958, I think will manifest again here? (wasn't able to figure out how to add a unit test)

Reply:
> We pass normalized source code to solc so source locations are relative to that instead of the original source code

Normalized here means in `Source::new` we strip '\r' from the source code